frontend/src/pages/Home/Home.tsx

```
@@ -16,34 +16,46 @@ async function fetchPokemons() {
 
 export const Home = () => {
   const [pokemonList, updatePokemonList] = React.useState<PokemonProps[]>([])
-  const isLoading = useRef(true)
+  const [isLoading, updateIsLoading] = React.useState(true)
+  const [requestSuccess, updateRequestSuccess] = React.useState(true)
 
   useEffect(() => {
     const fetchData = async () => {
       const pokemonData = await fetchPokemons()
-      isLoading.current = false
+      updateIsLoading(false)
       updatePokemonList(pokemonData)
     }
-    fetchData()
+
+    fetchData().catch(() => {
+      updateRequestSuccess(false)
+    })
   }, [])
 
   return (
     <div>
```

C'est un peu dur à lire toutes ces imbrications. Pour rendre ton code plus lisible tu peux utiliser ce qu'on appelle "fail fast".
ça consiste à remplacer une imbrication de if en commençant par le if qui fait s'arrêter le code. Dans ton cas ça donnerait un truc du genre

```
if(isLoading) {
   return <div>...</div>
}

if(isLoading) {
   return <div>Loading</div>
}
if(isError) {
   return <div>Error</div>
}
return <div>le cas classique</div>

```

>